### ISSUE TITLE
Clear definition between build and deploy stage

### ISSUE THREAD

**Issue by jamesla** [serverless/serverless#7464]
# Feature Proposal

## Description

When building packages using `sls package --package myPackage` it builds an artifact with variables baked in for the default stage (or alternatively a stage specified with -s).

This doesn't really make sense from a modern application standpoint. From a general build/deploy point of view, we build an artifact that isn't associated with any environment and then we deploy that artifact to each environment and pass in any environment specific variables at that point.

The advantage of doing it this way is that it simplifies CI/CD systems (since there isn't a build being run and an artefact being stored for each environment) as well as better dev/prod parity since the code bundled is identical as it transitions between environments.

Simply put we should be able to run:

On build server
```
sls package --package myPackage
zip -r app-1.0.0.zip myPackage
```

On deploy server
```
sls deploy --package ./app-1.0.0.zip -s dev
sls deploy --package ./app-1.0.0.zip -s prod
```

as clearly distinct stages.

Reading through the docs this didn't seem possible since when packaging you had to specificy a stage. Have other people ran into this? Otherwise is there a current way to get around this IE can we just package with a specific stage and then overwrite those staged variables at deploy time without any issues?

**Comment by jplock** [comment]
During our build process we build a package per environment (from the same source). Then for each environment we deploy the environment specific package.

**Comment by medikoo** [comment]
@jamesla great thanks for that report. We already received similar request, and discussing it at https://github.com/serverless/serverless/issues/6568

I'm closing this as duplicate, and please join a discussion there
